gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -568,13 +587,79 @@ var gpii = fluid.registerNamespace("gpii");
+     */
+    gpii.lifecycleManager.readPreference = function (that, preference, preferenceReadSuccessEvent, preferenceReadFailEvent) {
+        fluid.log("Lifecycle Manager: Received preference to read ", preference);
+        var session = that.getSession();
+
+        // Mock values for running preference thru matchMaker
+        var contextualPrefs = {};
+        var activeContextName = "gpii-default";
+        var prefSet = {};
+        fluid.each(preference, function (spec, pref) {
+            var prefPath = fluid.pathUtil.parseEL(pref);
+            fluid.set(contextualPrefs, prefPath, spec.value);
+        });
+        fluid.set(prefSet, ["contexts", activeContextName, "preferences"], contextualPrefs);
+
+        // Perform a local matchmaking to get matched solutions for the preference.
+        var matchMakingPromise = that.privateMatchMaker.doMatch("readSetting", prefSet);
+        matchMakingPromise.then(function (payload) {
+            fluid.set(payload, ["activeContextName"], activeContextName);
+            fluid.set(payload, ["preferences"], prefSet);
+            that.addLifecycleInstructionsToPayload(payload);
+
+            // compile a list of solution registry entries for all matched solutoins
+            var matchMakerSolutionRegistryEntries = fluid.get(payload, ["activeConfiguration", "lifecycleInstructions"]);
+            var settingsPromise = that.read(matchMakerSolutionRegistryEntries);
+            settingsPromise.then(function (allSettingValues) {
+                // As lifecycleManager.read() returns values for all settings that matched solutions support,
+                // only filter out setting values that are in the matchmaker result.
+                var matchMakerSettings = gpii.lifecycleManager.getMatchMakerSettings(allSettingValues, matchMakerSolutionRegistryEntries);
+
+                // Transform settings to preferences
+                var transformed = gpii.lifecycleManager.transformSettingsToPrefs(matchMakerSettings, matchMakerSolutionRegistryEntries);
+
+                // Filter matched preferences
+                var matchedPrefs = gpii.lifecycleManager.filterMatchedPrefs(transformed, contextualPrefs);
+
+                if (fluid.jQueryStandalone.isEmptyObject(matchedPrefs)) {
+                    preferenceReadFailEvent.fire();
+                } else {
+                    session.applier.change(["currentPreferences"], fluid.extend(true, fluid.copy(session.model.currentPreferences), matchedPrefs));
+                    preferenceReadSuccessEvent.fire();
+                }
+            });
+        });
+    };
+
+    /*
+     * Get preferences block for a certain context.
+     * @param {Object} contextualPrefs - A contextual preferences.
+     * @param {String} contextName - A context name.
+     * @return {Object} The actual preferences for the active context.
+     */
+    gpii.lifecycleManager.getContextedPreferences = function (contextualPrefs, contextName) {
```

Comment:
Given @klown 's https://github.com/GPII/universal/pull/823 is in progress, I think we should avoid introduce the name "context" in any new material in the repository just so we have less work when it comes time to reconcile all of this. Anticipating this work I think we should name this function something like "getPreferencesForSet" and change the name of argument 2 to "prefsSet". Argument 1 is also oddly named, it is not really the "contextual prefs" but actually the full prefs document.

Reply:
If the second argument is a name only, and not a set of preferences, then I suggest "prefsSetName".